src/client/automation/utils/utils.js

```diff
@@ -38,7 +53,13 @@ function setCaretPosition (element, caretPos) {
 }
 
 export function focusAndSetSelection (element, simulateFocus, caretPos) {
-    return new Promise(resolve => {
+    return new Promise(async resolve => {
+        // NOTE: Safari 13 blocks attempts to focus elements inside a third-party iframe before the user interacts with it
+        // https://developer.apple.com/documentation/safari-release-notes/safari-13-release-notes
+        // We can work around this restriction by focusing the <iframe> element beforehand
+        if (window.top !== window)
```

Comment:
Can we use `window.parent` instead of `window.top` here to check if we are in the iframe?

Reply:
I think we can, but `window.top` is used throughout the project.
https://github.com/DevExpress/testcafe/blob/43678b62c2a33685fbdd73dd9c4024c3c9ace776/src/client/automation/playback/scroll.js#L263
https://github.com/DevExpress/testcafe/blob/e7790ccad7a67b3962d7df16c117c88144faed04/src/client/automation/get-element.js#L85
Why is it better to use `window.parent`?

Reply:
In the next line we send a request to the window parent, but in the `if` condition we check the `window.top` property. I would prefer to have the same properties in the condition and in the request method. 
I understand that your case works correctly, so it's up to you.

Reply:
Personally, I think that we should keep using `window.top !== window` to check if we're in iframe for the consistency purposes (so that it's the same throughout the project).

I can change it to
```js
const isInIframe = window.top !== window;
if (isInIframe)
...
```
if this change makes it look better.

Alternatively, I could add the `isInIframe` function to `./src/utils` and use it to replace all similar conditions.

Reply:
It's a minor recommendation, so you can use the approach you prefer. I like the idea to extract the `isInIframe` function.